tests/TestHarness/Node.py

```diff
@@ -489,6 +490,34 @@ def waitForTransInBlock(self, transId, timeout=None):
         ret=Utils.waitForBool(lam, timeout)
         return ret
 
+    def checkBlockForTransactions(self, transIds, blockNum):
+        block = self.processCurlCmd("trace_api", "get_block", f'{{"block_num":{blockNum}}}', silentErrors=False, exitOnError=True)
+        if block['transactions']:
+            for trx in block['transactions']:
+                if trx['id'] in transIds:
+                    transIds.pop(trx['id'])
+        return transIds
+
+    def waitOnBlockTransactions(self, transIds, startBlock=2, timeout=30):
+        lastBlockProcessed = startBlock
+        finalBlock = sys.maxsize
+        if timeout is not None:
+            finalBlock = self.getHeadBlockNum() + timeout * 2
+        while len(transIds) > 0:
+            endBlock = self.getHeadBlockNum()
+            if endBlock >  finalBlock:
+                endBlock = finalBlock
+            for blockNum in range(endBlock, lastBlockProcessed - 1, -1):
+                transIds = self.checkBlockForTransactions(transIds, blockNum)
+                if len(transIds) == 0:
+                    return transIds
+            lastBlockProcessed = endBlock
+            if endBlock == finalBlock:
```

Comment:
Probably more accurate to use `self.getHeadBlockNum() >= finalBlock` instead of `endBlock` here in case enough time has passed that head has advanced.

Reply:
This falls into the same group as the previous comments. Additionally, the greater than is handled by lines 508 and 509 meaning it can never be greater than by this point in the code.

Reply:
I misunderstood timeout to be trying to constrain the amount of time spent in the waitOnBlockTransactions function call. I now understand it to be a book end on how many additional blocks this function will wait to be produced and audited for transactions before exiting and reporting not all trxs were found.